src/librustc_resolve/lib.rs

```diff
@@ -1226,6 +1229,8 @@ impl<'a> Resolver<'a> {
         invocations.insert(Mark::root(),
                            arenas.alloc_invocation_data(InvocationData::root(graph_root)));
 
+        let proc_macro_enabled = session.features.borrow().proc_macro;
```

Comment:
I think `#![feature(proc_macro)]` should imply `#![feature(use_extern_macros)]` to encourage importing attribute proc macros with `use`.
Additionally, we could warn against using `#[macro_use]` to import attribute proc macros.
cc @nrc